### ISSUE TITLE
validate input registration

### ISSUE THREAD

**Issue by trinhvandat** [tiangolo/fastapi#2112]
Hi guys!
I want to add validate input when user register. I don't know how can i do it? do you help me?

**Comment by ycd** [comment]
Yes i can definitely help about validating the input from user register, if you prove some minimal, reproducible example...

**Comment by trinhvandat** [comment]
Yes. Thanks for your reply.
I download folder keycloak-11.0.2 into keycloak.org website. I run keycloak server by command 
"./standalone.sh -Djboss.socket.binding.port-offset=100". It is running.

I create realm. Into realm, I create client, role, ....   But when I register user, the password field works fine even if I enter 1 character, this is not okay and I want to validate the password input field. Similar to other filed.

**Comment by ycd** [comment]
Can i see your endpoint and your Pydantic Model, it quite hard to help without them.

**Comment by trinhvandat** [comment]
Oh! Thanks for your reply. I am sorry. When I comment on github, i can't
add image. I will send image on here.

**Comment by Mause** [comment]
Feel free to paste the code into a comment with appropriate code formatting

**Comment by David-Lor** [comment]
> Yes. Thanks for your reply.
> I download folder keycloak-11.0.2 into keycloak.org website. I run keycloak server by command
> "./standalone.sh -Djboss.socket.binding.port-offset=100". It is running.
> 
> I create realm. Into realm, I create client, role, .... But when I register user, the password field works fine even if I enter 1 character, this is not okay and I want to validate the password input field. Similar to other filed.

So if I understand correctly, you want to interact with Keycloak API from your FastAPI app, and you want user given input data validated. You would have to create Pydantic models with the validation rules you want (refer to Pydantic documentation), and use these models as request body in your endpoints (refer to FastAPI documentation).

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 🍰 

@trinhvandat I'll close this issue, but if you still need help, please create a new issue following the template, answering the questions, and providing a minimal example that reproduces the problem. Thanks! ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
